spec/unit/file_system_spec.rb

```diff
@@ -25,14 +25,15 @@ def with_file_content(content)
   end
 
   context "#open" do
-    it "uses a default mode of 100644, the same as File.open, when specifying a nil mode" do
+    it "uses a default mode based on umask, the same as File.open, when specifying a nil mode" do
       file = tmpfile('file_to_update')
       expect(Puppet::FileSystem.exist?(file)).to be_falsey
 
       Puppet::FileSystem.open(file, nil, 'a') { |fh| fh.write('') }
 
-      # default mode of 0644 is applied
-      expect(File.stat(file).mode.to_s(8)).to eq('100644')
+      expected_perms = 666 - File.umask.to_s(8).to_i
+      # default mode is applied. 100 == 'regular file'
+      expect(File.stat(file).mode.to_s(8)).to eq("100#{expected_perms}")
```

Comment:
Hmm I'm on the fence.. can certainly update but I think it might make debugging a possible failure harder if we're not exposing these as octal - unless you know a way to keep these in octal representation without doing `to_s` ?